test/AbstractTransitionController.spec.ts

```diff
@@ -0,0 +1,209 @@
+      expect(app.transitionController.getTimelineDurationForComponent('[data-component="child-component-a"]')).to.be.a('number');
+    });
+
+    it('should try to getTimelineDurationForComponent by Component', () => {
+      expect(component.transitionController.getTimelineDurationForComponent(component)).to.be.a('number');
+    });
+
+    it('should try to getTimelineDurationForComponent by Element', () => {
+      expect(component.transitionController.getTimelineDurationForComponent(component.element)).to.be.a('number');
+    });
+
+    it('should try to getTimelineDurationForComponent with exact parameters', () => {
+      const reset = false;
+      const label = TransitionId[TransitionDirection.IN].TRANSITION_ID_1;
+      const direction = TransitionDirection.IN;
+
+      const spy = sinon.spy(component.transitionController, 'getTimelineInstance');
+      expect(component.transitionController.getTimelineDurationForComponent(component, direction, reset, label)).to.be.a('number');
+      expect(spy).to.be.calledWithExactly(component, direction, reset, label);
+      spy.restore();
+    });
+  });
+
+  describe('getTimelineInstance', () => {
+    it('should getTimelineInstance', () => {
+      const reset = false;
+      const labelOUT = TransitionId[TransitionDirection.OUT].TRANSITION_ID_1;
+
+      expect(component.transitionController.getTimelineInstance(component,  TransitionDirection.OUT, reset, labelOUT)).to.be.instanceOf(TimelineMax);
+    });
+  });
+
+
+  describe('getTimeline', () => {
+    it('should try to getTimeline by string', () => {
+      expect(app.transitionController.getTimeline('[data-component="child-component-a"]')).to.be.instanceOf(TimelineMax);
+    });
+
+    it('should try to getTimeline by Component', () => {
+      expect(component.transitionController.getTimeline(component)).to.be.instanceOf(TimelineMax);
+    });
+
+    it('should try to getTimeline by Element', () => {
+      expect(component.transitionController.getTimeline(component.element)).to.be.instanceOf(TimelineMax);
+    });
+
+    it('should try to getTimeline with exact parameters', () => {
+      const reset = false;
+      const label = TransitionId[TransitionDirection.IN].TRANSITION_ID_1;
+      const direction = TransitionDirection.IN;
+
+      const spy = sinon.spy(component.transitionController, 'getTimelineInstance');
+      expect(component.transitionController.getTimeline(component, direction, reset, label)).to.be.instanceOf(TimelineMax);
+      expect(spy).to.be.calledWithExactly(component, direction, reset, label);
+      spy.restore();
+    });
+  });
+
+  describe('parentController', () => {
+    it('should be the parent component', () => {
+      expect(component.transitionController.parentController).to.deep.equal(component);
+    });
+  });
+
+  describe('isHidden', () => {
+    it('should be a boolean', () => {
+      expect(component.transitionController.isHidden).to.be.a('boolean');
```

Comment:
It's better to check with true and false values (I believe the true was already tested earlier).